internal/core/metadata/uom/uom.go

```diff
@@ -0,0 +1,39 @@
+//
+// Copyright (C) 2022 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package uom
+
+type UnitsOfMeasureImpl struct {
+	Uom ConfigurationStruct
+}
+
+type ConfigurationStruct struct {
+	Source string
+	Units  map[string]Unit
+}
+
+type Unit struct {
+	Source string
+	Values []string
+}
+
+func (u *UnitsOfMeasureImpl) Validate(unit string) bool {
+	if u == nil {
+		return false
+	}
+	if unit == "" || len(u.Uom.Units) == 0 {
```

Comment:
IMO, If no UOM.Units specified at all, then nothing to validate against, so should always pass.

```suggestion
	if len(u.Uom.Units) == 0 {
```

Reply:
I think we should still keep the first check otherwise validation will fail if user doesn't provide the `units` property ?

Reply:
Ahh , yep. I missed the OR part of the conditional.